File: model/recipe.go
package model

type Recipes struct {
	Objects []*Recipe `json:"recipes"`
}

type Recipe struct {
	ID                 int64         `json:"id"`
	Name               string        `json:"name"`
	Description        string        `json:"description"`
	Url                string        `json:"url"`
	Image              string        `json:"image"`
	PrepTime           string        `json:"prepTime"`
	CookTime           string        `json:"cookTime"`
	TotalTime          string        `json:"totalTime"`
	RecipeCategory     string        `json:"recipeCategory"`
	Keywords           string        `json:"keywords"`
	RecipeYield        int           `json:"recipeYield"`
	Tool               []string      `json:"tool"`
	RecipeIngredient   []string      `json:"recipeIngredient"`
	RecipeInstructions []string      `json:"recipeInstructions"`
	Nutrition          *NutritionSet `json:"nutrition"`
	DateModified       string        `json:"dateModified"`
	DateCreated        string        `json:"dateCreated"`
}

type NutritionSet struct {
	Calories     string `json:"calories"`
	Carbohydrate string `json:"carbohydrateContent"`
	Fat          string `json:"fatContent"`
	SaturatedFat string `json:"saturatedFatContent"`
	Cholesterol  string `json:"cholesterolContent"`
	Protein      string `json:"proteinContent"`
	Sodium       string `json:"sodiumContent"`
	Fiber        string `json:"fiberContent"`
	Sugar        string `json:"sugarContent"`
}

func (r *Recipe) IsCreatedSameTime(other *Recipe) bool {
	return r.DateCreated == other.DateCreated
}

func (r *Recipe) IsModified(other *Recipe) bool {
	return r.DateModified != other.DateModified
}

File: web/flutter-app/lib/src/models/recipe.dart
import 'package:equatable/equatable.dart';

import 'nutrition.dart';

class RecipeModel extends Equatable {
  final int? id;
  final String? name;
  final String? description;
  final String? category;
  final String? url;
  final String? image;
  final String? prepTime;
  final String? cookTime;
  final String? totalTime;
  final String? keywords;
  final int? recipeYield;
  final List<String?> tool;
  final List<String?> recipeIngredients;
  final List<String?> recipeInstructions;
  final NutritionModel? nutrition;
  final String? dateModified;
  final String? dateCreated;

  RecipeModel({
    this.id,
    this.name,
    this.description,
    this.category,
    this.url,
    this.image,
    this.prepTime,
    this.cookTime,
    this.totalTime,
    this.keywords,
    this.recipeYield,
    required this.tool,
    required this.recipeIngredients,
    required this.recipeInstructions,
    this.nutrition,
    this.dateModified,
    this.dateCreated,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        category,
        url,
        image,
        prepTime,
        cookTime,
        totalTime,
        keywords,
        recipeYield,
        tool,
        recipeIngredients,
        recipeInstructions,
        nutrition,
        dateModified,
        dateCreated,
      ];

  RecipeModel.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        description = json['description'],
        category = json['recipeCategory'],
        url = json['url'],
        image = json['image'],
        prepTime = json['prepTime'],
        cookTime = json['cookTime'],
        totalTime = json['totalTime'],
        keywords = json['keywords'],
        recipeYield = json['recipeYield'],
        tool = json['tool'] ?? [],
        recipeIngredients = (json['recipeIngredient'] as List).cast<String>(),
        recipeInstructions =
            (json['recipeInstructions'] as List).cast<String>(),
        nutrition = NutritionModel.fromJson(json['nutrition']),
        dateModified = json['dateModified'],
        dateCreated = json['dateCreated'];

  String totalTimeToString() {
    if (totalTime == "" || totalTime == "PT0H0M") {
      return "0m";
    }

    var parts = totalTime!.split('PT')[1].split('H');
    var hour = parts[0];
    var min = parts[1].split('M')[0];
    return '${hour}h${min}m';
  }
}
